src/ultk/language/grammar/grammar.py

```diff
@@ -165,6 +165,21 @@ def add_child(self, child) -> None:
         else:
             self.children = self.children + (child,)
 
+    def complement(self) -> Meaning:
+        """Get the complement of the meaning of this expression, i.e. the set of all referents for which
+        the expression evaluates to False."""
+
+        return Meaning(
```

Comment:
I think this is out of date with the new `Meaning`s that are mappings, not just sets of referents.  And also, if we do want this, it should be with `Expression`, not just `GrammaticalExpression`, since nothing depends on the grammatical structure